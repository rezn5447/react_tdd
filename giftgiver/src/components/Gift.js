import React, { Component } from 'react';
import { Row, Input, Button, Icon } from 'react-materialize';

class Gift extends Component {
  constructor() {
    super();
    this.state = { person: '', present: '' };
  }
  render() {
    return (
      <Row>
        <Input
          s={6}
          className="input-person"
          type="text"
          label="person"
          onChange={event => this.setState({ person: event.target.value })}
        >
          <Icon>account_circle</Icon>
        </Input>
        <Input
          s={6}
          className="input-present"
          type="text"
          label="present"
          onChange={event => this.setState({ present: event.target.value })}
        >
          <Icon>card_giftcard</Icon>
        </Input>
        <Button
          className="btn-remove"
          onClick={() => this.props.removeGift(this.props.gift.id)}
        >
          Remove Gift {this.props.gift.id}
        </Button>
      </Row>
    );
  }
}

export default Gift;
