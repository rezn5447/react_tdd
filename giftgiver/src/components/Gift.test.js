import React from 'react';
import { shallow } from 'enzyme';
import Gift from './Gift';

describe('Gift Component', () => {
  const gift = shallow(<Gift />);

  it('renders correctly', () => {
    expect(gift).toMatchSnapshot();
  });
});
