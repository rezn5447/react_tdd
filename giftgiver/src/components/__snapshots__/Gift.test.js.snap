// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gift
    gift={
        Object {
            "id": 1,
          }
    }
    removeGift={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Input
          checked={false}
          className="input-person"
          label="person"
          onChange={[Function]}
          s={6}
          type="text"
>
          <Icon>
                    account_circle
          </Icon>
</Input>,
        <Input
          checked={false}
          className="input-present"
          label="present"
          onChange={[Function]}
          s={6}
          type="text"
>
          <Icon>
                    card_giftcard
          </Icon>
</Input>,
        <Button
          className="btn-remove"
          node="button"
          onClick={[Function]}
>
          Remove Gift 
          1
</Button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "checked": false,
          "children": <Icon>
            account_circle
</Icon>,
          "className": "input-person",
          "label": "person",
          "onChange": [Function],
          "s": 6,
          "type": "text",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "account_circle",
          },
          "ref": null,
          "rendered": "account_circle",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "checked": false,
          "children": <Icon>
            card_giftcard
</Icon>,
          "className": "input-present",
          "label": "present",
          "onChange": [Function],
          "s": 6,
          "type": "text",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "card_giftcard",
          },
          "ref": null,
          "rendered": "card_giftcard",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Remove Gift ",
            1,
          ],
          "className": "btn-remove",
          "node": "button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Remove Gift ",
          1,
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Input
            checked={false}
            className="input-person"
            label="person"
            onChange={[Function]}
            s={6}
            type="text"
>
            <Icon>
                        account_circle
            </Icon>
</Input>,
          <Input
            checked={false}
            className="input-present"
            label="present"
            onChange={[Function]}
            s={6}
            type="text"
>
            <Icon>
                        card_giftcard
            </Icon>
</Input>,
          <Button
            className="btn-remove"
            node="button"
            onClick={[Function]}
>
            Remove Gift 
            1
</Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": false,
            "children": <Icon>
              account_circle
</Icon>,
            "className": "input-person",
            "label": "person",
            "onChange": [Function],
            "s": 6,
            "type": "text",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "account_circle",
            },
            "ref": null,
            "rendered": "account_circle",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": false,
            "children": <Icon>
              card_giftcard
</Icon>,
            "className": "input-present",
            "label": "present",
            "onChange": [Function],
            "s": 6,
            "type": "text",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "card_giftcard",
            },
            "ref": null,
            "rendered": "card_giftcard",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Remove Gift ",
              1,
            ],
            "className": "btn-remove",
            "node": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Remove Gift ",
            1,
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
